plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -388,17 +389,36 @@ public void testCaseSensitiveAggregationPushdown()
                     supportsPushdownWithVarcharInequality && supportsCountDistinctPushdown,
                     node(ExchangeNode.class, node(AggregationNode.class, anyTree(node(TableScanNode.class)))))
                     .skippingTypesCheck()
-                    .matches("VALUES (BIGINT '4', BIGINT '4')");
+                    .matches("VALUES (BIGINT '4', BIGINT '3')");
 
             assertConditionallyPushedDown(getSession(),
                     "SELECT count(DISTINCT a_char), count(DISTINCT a_bigint) FROM " + table.getName(),
                     supportsPushdownWithVarcharInequality && supportsCountDistinctPushdown,
                     node(ExchangeNode.class, node(AggregationNode.class, anyTree(node(TableScanNode.class)))))
                     .skippingTypesCheck()
-                    .matches("VALUES (BIGINT '4', BIGINT '4')");
+                    .matches("VALUES (BIGINT '4', BIGINT '3')");
+
+            assertConditionallyPushedDown(getSession(),
+                    "SELECT count(DISTINCT a_string), sum(DISTINCT a_bigint) FROM " + table.getName(),
+                    supportsPushdownWithVarcharInequality && supportsSumDistinctPushdown,
+                    node(ExchangeNode.class, node(AggregationNode.class, anyTree(node(TableScanNode.class)))))
+                    .skippingTypesCheck()
+                    .matches(sumDistinctAggregationPushdownExpectedResult());
```

Comment:
do you still need this overridable part after adding `skippingTypesCheck`?

Reply:
perhaps decimal 8 doesn't come back as a Long

Reply:
to avoid having `sumDistinctAggregationPushdownExpectedResult` method, we can cast `sum` result back to `bigint` in the test query

Reply:
@findepi Yes, you are right, but currently I do not have any clearly path to make it on Oracle, do you have any thoughts?

Reply:
`CAST` will prevent pushdown @findepi.

Instead of method to return entire result let's just allow choosing what data-type to return. With override which returns entire result the risk is that the result values may not match super and mask correctness issues.

Reply:
ignore me, the CAST on result of aggregation doesn't prevent pushdown. So the CAST approach should work.

Reply:
@hashhar Could share what's the test sql? I saw the `plan not match` on my local